HPCC-24341 Dismiss dialog and clear fields

miguelvazq opened the pull request:
- Dismissed dialog due to an undefined widget
- Cleared input fields if passwords did not meet criteria

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Process of PR-13919, label: HPCC-24341 starts now.
The reason of this test is: Did not build yet.
Commit ID: aa537c11bf5aaa18fc2687a6be5b13f523357a04
Estimated completion time is ~0.2 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13919, label: HPCC-24341 starts now.
The reason of this test is: Forced to re-test.
Commit ID: aa537c11bf5aaa18fc2687a6be5b13f523357a04
Estimated completion time is ~0.1 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-139.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-139.ca-central-1.compute.internal
Sha: aa537c11bf5aaa18fc2687a6be5b13f523357a04
Build: success
Milestone:Install hpccsystems-platform-community_7.10.1-closedown0.el7.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  724 sec (00:12:04) |  117 sec (00:01:57) |   23 sec (00:00:23) |   17 sec (00:00:17) |    0 sec (00:00:00) |   15 sec (00:00:15) |  905 sec (00:15:05) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: aa537c11bf5aaa18fc2687a6be5b13f523357a04
Build: success
Milestone:Install hpccsystems-platform-community_7.10.1-closedown0.el7.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   14 sec (00:00:14) |  998 sec (00:16:38) |  123 sec (00:02:03) |   23 sec (00:00:23) |   17 sec (00:00:17) |    0 sec (00:00:00) |   15 sec (00:00:15) | 1190 sec (00:19:50) |

miguelvazq commented on esp/src/eclwatch/CurrentUserDetailsWidget.js:
```diff
@@ -51,7 +52,9 @@ define([
         //  Hitched actions  ---
         _onSave: function (event) {
-            var dialog = dijit.byId("stubUserDialog");
+            var context = this;
```
The majority of the work is done here where I pass in the widget as a param when initialized since it is undefined. As well as clearing the fields if exception is thrown.

GordonSmith reviewed (commented):
Did HPCCPlatformWidget.js actually change - lots of white space changes?

miguelvazq commented:
> Did HPCCPlatformWidget.js actually change - lots of white space changes?

Yes, the eslint format on save must have fired off those changes.

miguelvazq commented on esp/src/eclwatch/HPCCPlatformWidget.js:
```diff
@@ -344,67 +349,70 @@ define([
-            if (!userInfo.init({ Username: this.userName })) {
+            if (!userInfo.init({
+                    Username: this.userName,
+                    Widget: userDialog
```
Pass in param here.

GordonSmith commented on esp/src/eclwatch/templates/CurrentUserDetailsWidget.html:
```diff
@@ -27,16 +29,22 @@ <h2>
-                            <input name="newPasswordRetype" type="password" required="true" pwtype="verify" data-dojo-props="invalidMessage:'${i18n.PasswordsDoNotMatch}', placeHolder:'${i18n.MustContainUppercaseAndSymbol}'" data-dojo-type="dijit.form.ValidationTextBox" />
+                            <input name="newPasswordRetype" type="password" required="true" pwtype="verify"
+                                data-dojo-props="invalidMessage:'${i18n.PasswordsDoNotMatch}', placeHolder:'${i18n.MustContainUppercaseAndSymbol}'"
+                                data-dojo-type="dijit.form.ValidationTextBox" />
```
Looks like you have enabled a HTML formatter?  Can you roll back these changes (or adjust your formatter to not split the lines?

Commit:
HPCC-24341 Dismiss dialog and clear fields

- Dismissed dialog due to an undefined widget
- Cleared input fields if passwords did not meet criteria

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

miguelvazq commented:
@GordonSmith modified IDE to not format on save for selected options.

HPCCSmoketest commented:
Process of PR-13919, label: HPCC-24341 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 5c14805b65c37a6ab4689f719b2cc5aaa21b2000
Estimated completion time is ~0.1 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-125.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13919, label: HPCC-24341 starts now.
The reason of this test is: New commit.
Commit ID: 5c14805b65c37a6ab4689f719b2cc5aaa21b2000
Estimated completion time is ~0.2 hour(s)
centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-125.ca-central-1.compute.internal
Sha: 5c14805b65c37a6ab4689f719b2cc5aaa21b2000
Build: success
Milestone:Install hpccsystems-platform-community_7.10.3-closedown0.el7.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  745 sec (00:12:25) |  118 sec (00:01:58) |   22 sec (00:00:22) |   20 sec (00:00:20) |    0 sec (00:00:00) |   14 sec (00:00:14) |  928 sec (00:15:28) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.8.2003 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 5c14805b65c37a6ab4689f719b2cc5aaa21b2000
Build: success
Milestone:Install hpccsystems-platform-community_7.10.3-closedown0.el7.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   13 sec (00:00:13) | 1036 sec (00:17:16) |  123 sec (00:02:03) |   23 sec (00:00:23) |   17 sec (00:00:17) |    0 sec (00:00:00) |   16 sec (00:00:16) | 1228 sec (00:20:28) |

GordonSmith commented on esp/src/eclwatch/CurrentUserDetailsWidget.js:
```diff
@@ -51,7 +53,9 @@ define([
         _onSave: function (event) {
-            var dialog = dijit.byId("stubUserDialog");
+            var context = this;
+            var dialog = this.params.Widget;
```
Passing an entire Widget via params will mean that this page cannot be displayed on its own or in a react component (which may be ok?)

miguelvazq replied:
I envision us creating a dialog component that we pass in ...props to replace items such as these in the current world.

GordonSmith reviewed (approved):
Looks ok - one question re the potential reuse?

GordonSmith commented:
@ghalliday good to pull

ghalliday commented:
Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>
